### PR TITLE
Remove measure APIs from amp-auto-lightbox

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #31540.
Partial for #33678.

This blocks #33649.

### CODE REVIEW

File: extensions/amp-auto-lightbox/0.1/measure.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+/** @type {!WeakMap<!Element, !Deferred>} */
+const measuringElements = new WeakMap();
+
+/**
+ * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+ * @param {!Element} element
+ * @return {!Promise<!ClientRect>}
+ */
+export function measure(ampdoc, element) {
```
Comment (jridgewell) [inline extensions/amp-auto-lightbox/0.1/measure.js:30]: Should we promote this to `src/dom.js`?

**Review by jridgewell — APPROVED** [review]

File: extensions/amp-auto-lightbox/0.1/measure.js (hunk 1 above)
Comment (dvoytenko) [reply extensions/amp-auto-lightbox/0.1/measure.js:30]: Sorry, was going to comment on that and forgot. @samouri has already created `src/utils/intersection` and I can reuse that. The only nit is that the `src/utils/intersection` uses an `IntersectionObserver{root:document}`, which triggers the polyfill path in more cases than a simple `{root:null}` that I do here. So I wasn't sure if it's worth reusing the existing util or rolling this as a single-use utility for the extension. LMK what you think.

**Review by samouri — APPROVED** [review]
I'm okay with whichever way you decide:
1. Add a TODO comment re. merging the helper with `measureIntersection` once iOS support for document as root improves
2. and/or promoting the helper to a utils area

**Comment by dvoytenko** [comment]
> I'm okay with whichever way you decide:
> 
> 1. Add a TODO comment re. merging the helper with `measureIntersection` once iOS support for document as root improves
> 2. and/or promoting the helper to a utils area

I did both: promoted it to `src/utils` and left a `TODO(#33678)` to cleanup utils once the `IntersectionObserver{root:document}` is better supported.

**Pull request merged**